### PR TITLE
Help trace worker crash in Kusto.

### PR DESCRIPTION
**Pull request by TingluoHuang**
There was a bug in the Actions hosted runner virtual environment that causes the available disk space reduced. http://github.com/actions/virtual-environments/issues/709

Some customer's workflow used all disk space and cause the runner crash as well.

We want to get alert when the number of worker crash increase, so we can proactively investigate these issue instead of waiting for customer report.

Each worker crash means there is a failed workflow run and the customer is not happy with it.

### CODE REVIEW

File: src/Runner.Listener/JobDispatcher.cs [hunk 1]
```diff
@@ -952,8 +951,10 @@ private async Task LogWorkerProcessUnhandledException(Pipelines.AgentJobRequestM
                 ArgUtil.NotNull(timeline, nameof(timeline));
                 TimelineRecord jobRecord = timeline.Records.FirstOrDefault(x => x.Id == message.JobId && x.RecordType == "Job");
                 ArgUtil.NotNull(jobRecord, nameof(jobRecord));
+                var unhandledExceptionIssue = new Issue() { Type = IssueType.Error, Message = errorMessage };
+                unhandledExceptionIssue.Data["_worker_crash_unhandled_exception"] = string.Empty;
```
Comment (TingluoHuang) [inline src/Runner.Listener/JobDispatcher.cs:955]: service will check the issue data to decide whether fire product trace event.

File: src/Runner.Listener/JobDispatcher.cs [hunk 2]
```diff
@@ -952,8 +951,10 @@ private async Task LogWorkerProcessUnhandledException(Pipelines.AgentJobRequestM
                 ArgUtil.NotNull(timeline, nameof(timeline));
                 TimelineRecord jobRecord = timeline.Records.FirstOrDefault(x => x.Id == message.JobId && x.RecordType == "Job");
                 ArgUtil.NotNull(jobRecord, nameof(jobRecord));
+                var unhandledExceptionIssue = new Issue() { Type = IssueType.Error, Message = errorMessage };
+                unhandledExceptionIssue.Data[Constants.Runner.WorkerCrashIssueDataKey] = string.Empty;
```
Comment (TingluoHuang) [inline src/Runner.Listener/JobDispatcher.cs:955]: service will check the issue data to decide whether fire product trace event.

File: src/Runner.Worker/ActionCommandManager.cs [hunk 3]
```diff
@@ -486,7 +486,10 @@ public void ProcessCommand(IExecutionContext context, string inputLine, ActionCo
 
             foreach (var property in command.Properties)
             {
-                issue.Data[property.Key] = property.Value;
+                if (!string.Equals(property.Key, Constants.Runner.WorkerCrashIssueDataKey, StringComparison.OrdinalIgnoreCase))
```
Comment (ericsciple) [inline src/Runner.Worker/ActionCommandManager.cs:489]: dont we already filter everything other than specific properties?

File: src/Runner.Listener/JobDispatcher.cs (hunk 2 above)
Comment (ericsciple) [reply src/Runner.Listener/JobDispatcher.cs:955]: Think about:

Should we make this more generic? Do we have any other cases that warrant a more generic telemetry mechanism, e.g. `[...].Data["_telemetry"] = WORKER_CRASH;<MESSAGE>`

^ That way we can start adding TYPE;MESSAGE more generically to issues?

**Pull request merged**